src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -480,6 +499,22 @@ real :: RoScl     ! The scaling function for MEKE source term
       enddo ; enddo
     endif
 
+    if (CS%no_slip) then
+      do J=js-2,Jeq+1 ; do I=is-2,Ieq+1
+        vort_xy(I,J) = (2.0-G%mask2dBu(I,J)) * ( dvdx(I,J) - dudy(I,J) )
+      enddo ; enddo
+    else
+      do J=js-2,Jeq+1 ; do I=is-2,Ieq+1
+        vort_xy(I,J) = G%mask2dBu(I,J) * ( dvdx(I,J) - dudy(I,J) )
+      enddo ; enddo
+    endif
+
+! Vorticity gradient
+   do J=js-1,Jeq+1 ; do I=is-1,Ieq+1
+      vort_xy_dx(i,J) = CS%DY_dxBu(I,J)*(vort_xy(I,J)*G%IdyCu(I,j) - vort_xy(I-1,J)*G%IdyCu(I-1,j))
```

Comment:
Because vort_xy_dx and vort_xy_dy have different valid sizes and are used below with different stencils, they can not be calculated in the same loop-blocks, as is occurring in line 513.  Please compare these loop extents with the extents used below in lines 552 and 679 (keeping in mind that in symmetric mode Jsq=js-1) to make sure there are no uninitialized values that are being used. (It looks to me like there will be a problem in symmetric memory mode.)